src/lib/containers/SynapseTable.tsx

```diff
@@ -248,6 +259,67 @@ export default class SynapseTable extends React.Component<QueryWrapperChildProps
     )
   }
 
+  private showGroupRowData = (selectedRow: Row) => (_event: React.MouseEvent<HTMLButtonElement>) => {
+    // magic happens - parse query, deep copy query bundle request, modify, encode, send to Synapse.org.  Easy!
+    const queryCopy = cloneDeep(this.props.getLastQueryRequest!().query)
+    // unpack all the data
+    const { data } = this.props
+    const { queryResult, columnModels } = data!
+    const { queryResults } = queryResult
+    const { headers } = queryResults
+    const parsed = this.getSqlUnderlyingDataForRow(
+      selectedRow,
+      queryCopy.sql,
+      headers,
+      columnModels)
+    queryCopy.sql = parsed.newSql
+    const queryJSON = JSON.stringify(queryCopy)
+    // encode this copy of the query (json)
+    const encodedQuery = btoa(queryJSON)
+    // open this in a new window on synapse.org
+    window.open(`https://www.synapse.org/#!Synapse:${parsed.synId}/tables/query/${encodedQuery}`, '_blank')
+  }
+
+  public getSqlUnderlyingDataForRow(
+    selectedRow: Row,
+    originalSql: string,
+    headers: SelectColumn[],
+    columnModels: ColumnModel[]
+  ): { synId: string, newSql: string } {
+    const tokens: string[][] = lexer.tokenize(originalSql)
+    // remove all tokens after group (by truncating array)
+    tokens.length = tokens.findIndex(el => el[0] === 'GROUP')
```

Comment:
instead of truncating with length, you could instead use the slice method - 
 
tokens = tokens.slice(0, tokens.findIndex(el => el[0] === 'GROUP'))

Reply:
Read somewhere that truncating using the length parameter was faster, but
1.  Using slice is more clear IMO
2.  For this particular function, we're calling it once, so even if it was faster we should not sacrifice readability.
Made the suggested change!